test/unit/proxy_test.rb

```diff
@@ -483,15 +483,8 @@ def test_apicast_configuration_driven
                               api_backend: 'http://example.com',
                               api_test_path: '/path',
                               apicast_configuration_driven: false)
-    ::ApicastV1DeploymentService.any_instance.expects(:deploy).with(proxy).returns(true)
-
-    analytics.expects(:track).with('Sandbox Proxy Deploy', success: true)
-    analytics.expects(:track).with('Sandbox Proxy updated',
-                                   api_backend: 'http://example.com:80',
-                                   api_test_path: '/path',
-                                   success: true)
-    analytics.expects(:track).with('APIcast Hosted Version Change', {enabled: false, service_id: proxy.service_id, deployment_option: 'hosted'})
-    Logic::RollingUpdates.stubs(skipped?: true)
+    # TODO: remove this when ApicastV1DeploymentService class is removed
+    ::ApicastV1DeploymentService.any_instance.expects(:deploy).with(proxy).returns(true).never
```

Comment:
I don't think we can forget removing it as it will raise, so maybe the `TODO` comment not needed.

Reply:
now we are validating only apicast_configuration_driven -> true or false
```
   if apicast_configuration_driven?
      deploy_v2
   else # this is service mesh (and "future"others that do not have staging)
      deploy_v2 && deploy_production_v2
   end
```